internal/app/app.go

```diff
@@ -178,3 +200,80 @@ func loadChains(cfg config.NeutronQueryRelayerConfig, logRegistry *nlogger.Regis
 
 	return neutronChain, targetChain, nil
 }
+
+type connectionParams struct {
+	neutronChainID  string
+	neutronClientID string
+
+	targetChainID      string
+	targetClientID     string
+	targetConnectionID string
+}
+
+func loadConnParams(ctx context.Context, neutronClient, targetClient *rpcclienthttp.HTTP, restAddress string, neutronConnectionId string, logger *zap.Logger) (*connectionParams, error) {
+	restClient, err := raw.NewRESTClient(restAddress)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get newRESTClient: %w", err)
+	}
+
+	targetStatus, err := targetClient.Status(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch target chain status: %w", err)
+	}
+
+	neutronStatus, err := neutronClient.Status(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch neutron chain status: %w", err)
+	}
+
+	// NOTE: waiting for connection to be created
```

Comment:
Why do we need to wait for the connection to be created?
I mean, when we start an ICQ relayer instance we expect that the necessary connection is created already, don't we?

Reply:
Normally yes, but in case of integration tests no

Reply:
this can be treat as connection health check, i.e. make sure the chain is sound and the address is configured properly

Reply:
what I only don't like here is:
1. this `NOTE:` — cmon it's a comment and therefore it's already a note, why leave such an obvious explanation;
2. the `client` variable name is deceptive: it's not a client, but a REST call response.